platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -49,127 +49,92 @@
-    if (zookeeperHosts == null) {
+
+    if (StringUtils.isBlank(zookeeperHosts)) {
       LOGGER.warn(
           "Cannot create Solr Cloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      return null;
+      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
     }
-    return getClient(zookeeperHosts, core);
+    LOGGER.debug(
+        "Solr({}): Creating a Solr Cloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
+    return new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
   }
 
-  /**
-   * Creates a new {@link CloudSolrClient} using the list of Zookeeper hosts and collection name
-   * provided.
-   *
-   * @param zookeeperHosts comma-separate list of Zookeeper hosts managing the Solr Cloud
-   *     configuration
-   * @param collection name of the collection to create
-   * @return {@code Future} used to retrieve the new {@link CloudSolrClient} instance
-   */
-  public static Future<SolrClient> getClient(String zookeeperHosts, String collection) {
-    RetryPolicy retryPolicy =
-        new RetryPolicy()
-            .retryWhen(null)
-            .withBackoff(10, TimeUnit.MINUTES.toMillis(1), TimeUnit.MILLISECONDS);
-    return Failsafe.with(retryPolicy)
-        .with(EXECUTOR_SERVICE)
-        .onRetry(
-            (c, failure, ctx) ->
-                LOGGER.debug(
-                    "Attempt {} failed to create Solr Cloud client for collection {} using Zookeeper hosts [{}]. Retrying again.",
-                    ctx.getExecutions(),
-                    collection,
-                    zookeeperHosts))
-        .onFailedAttempt(
-            failure ->
-                LOGGER.debug(
-                    "Attempt failed to create Solr Cloud client for collection {} using Zookeeper hosts [{}]",
-                    collection,
-                    zookeeperHosts,
-                    failure))
-        .onSuccess(
-            client ->
-                LOGGER.debug(
-                    "Successfully created Solr Cloud client for collection {}", collection))
-        .onFailure(
-            failure ->
-                LOGGER.warn(
-                    "All attempts failed to create Solr Cloud client for collection {} using Zookeeper host [{}]",
-                    collection,
-                    zookeeperHosts,
-                    failure))
-        .get(() -> createSolrCloudClient(zookeeperHosts, collection));
-  }
-
-  private static ScheduledExecutorService createExecutorService() throws NumberFormatException {
-    Integer threadPoolSize =
-        NumberUtils.toInt(
-            System.getProperty("org.codice.ddf.system.threadPoolSize"), THREAD_POOL_DEFAULT_SIZE);
-    return Executors.newScheduledThreadPool(
-        threadPoolSize,
-        StandardThreadFactoryBuilder.newThreadFactory("solrCloudClientFactoryThread"));
-  }
+  @VisibleForTesting
```

Comment:
I marked it `@VisibleForTesting` because it is tested directly. I am testing it directly because it is being passed as a supplier to the `SolrClientAdapter` constructor so I feel it is an internal API worth being tested on its own. In addition, the annotation does shut up static analysis and does provide feedback to the developer to indicate that it is an internal method being tested.